verified_boot.md

```diff
@@ -0,0 +1,240 @@
+class: center, middle, intro
+
+# Verified Boot and firmware updates
+
+## How to do them securely and openly
+
+.center[<img src="/remark-templates/dasharo-presentation-template/images/dasharo-sygnet-white.svg" width="150px" style="margin-left:-20px">]
+
+???
+
+<!--
+SPDX-FileCopyrightText: 2024 3mdeb <contact@3mdeb.com>
+
+SPDX-License-Identifier: CC-BY-SA-4.0
+-->
+
+---
+
+# whoami
+
+- Michał Kopeć
+- Firmware Engineer at 3mdeb since 2021
+- Develops Dasharo for laptops, network appliances and other platforms
+- Uses Arch btw
+
+---
+
+# Agenda
+
+- Verified Boot
+  - in Chrome
+  - in Dasharo
+- Problem statement
+- Boot Guard
+- CBFS verification
+- Secure firmware updates
+
+---
+
+# Verified boot
+
+- Verified Boot: Cryptographic verification of the boot process to ensure only
+  code from a trusted source (e.g. device vendor) can run
+- coreboot supports Vboot, Google's version of verified boot scheme
+- Vboot has:
+  - A root of trust
+  - Firmware verification and measurements to TPM
+  - Signatures and public tools for self-signing
+  - A special A/B update mechanism
+  - Support for re-ownership
+- Dasharo uses Vboot
+
+???
+
+- Google created Vboot for chrome devices
+- In Chrome, vboot is also used for OS and EC secure boot
+- We use some subset of Vboot's capabilities
+
+---
+
+# Verified Boot in Chrome devices
+
+.center.image-55[![](/img/vboot/cros.svg)]
+
+- Write protection: WP# pin on the BIOS chip
+  - WP# pin is controlled by Cr50 security chip (newer devices) or physical
+    screw (older devices)
+- Updates: An OS service handles the A/B update scheme
+  - One slot is updated, and when confirmed bootable, the other slot is updated
+    too
+- Write protected portion of the flash is **not** updated, and serves as
+  recovery
+  - WP region contains the initial bootblock and verifies the A/B slots
+  - When both slots fail or recovery is manually requested (e.g. by keyboard
+    key), the firmware boots from the recovery partition
+
+???
+
+- Cr50 is a Google's TPM-like device, also known as Titan C
+
+---
+
+# Verified Boot in Dasharo
+
+.center.image-20[![](/remark-templates/dasharo-presentation-template/images/dasharo-sygnet.svg)]
+
+- Write protection is provided by chipset
```

Comment:
can be also provided by SPI chip, but it is HW dependent to have a WP pin exposed. Also SPI chip protection has restrictions on the regions that can be protected

Reply:
https://github.com/Dasharo/presentations/pull/60/files#diff-936438ab5cbf8aeccc31d7b1363cc502692b130865656d9113a709b30176076aR87